pkg/web/apps/apps_service.go

```diff
@@ -53,3 +55,77 @@ func (a *appsService) GetAppByID(id string) (*models.App, error) {
 
 	return app, nil
 }
+
+// InitClientApp retrieves the list of apps from the repository
+func (a *appsService) InitClientApp(deviceInfo *models.Device) (*models.InitClient, error) {
+	// Check if the app exists in the database for the sent app_id
+	if _, err := a.repository.GetAppByAppID(deviceInfo.AppID); err != nil {
+		return nil, err
+	}
+
+	version, err := a.repository.GetVersionByAppIDAndVersion(deviceInfo.AppID, deviceInfo.Version)
+
+	// If any error other Not Found error occurred, return
+	if err != nil && err != models.ErrNotFound {
+		return nil, err
+	}
+
+	// If the version does not exist, create it
+	if err == models.ErrNotFound {
+		// Create new uuid for our new app version
+		versionUUID := uuid.New()
+
+		version = &models.Version{
+			ID:               versionUUID.String(),
+			Version:          deviceInfo.Version,
+			AppID:            deviceInfo.AppID,
+			NumOfAppLaunches: 0,
+		}
+	}
+
+	// Increment the version App Launches
+	version.NumOfAppLaunches++
+
+	// Update the existing version or create a new one
+	if err := a.repository.InsertVersionOrUpdateNumOfAppLaunches(version); err != nil {
+		log.Error(err)
+		return nil, err
+	}
+
+	// If we can't find the device by version and app ID
+	if _, err = a.repository.GetDeviceByVersionAndAppID(version.Version, deviceInfo.AppID); err == models.ErrNotFound {
```

Comment:
The logic for these device calls should be changed. The flow should be as follows:

![device logic](https://user-images.githubusercontent.com/1596014/53734270-66511480-3e7b-11e9-87a3-bd027f7e8a02.png)

Reply:
@davidffrench The version ID is not currently being sent in the `/init` POST,  so the init device and db device cannot be compared on this property. Am I missing something?

Reply:
it is not sent in the `/init` endpoint but is included in the version retrieved from the database. The property called `version` will have a `id` property you can check against.

Reply:
@craicoverflow please check the suggestion https://github.com/aerogear/mobile-security-service/pull/76#discussion_r

Reply:
@davidffrench the version of some app should not be inserted by the init call. If we do it we will create any version send from the SDK. Is it really makes sense? It is something that I do not remember we discussed about, however, shows more one action that will be trigged by the OCP. How a developer deploy a new client version in Aerogear? In RHMAP it is done in the build tab of the client app. c/c @craicoverflow